engine/apps/public_api/views/schedules.py

```diff
@@ -120,3 +128,84 @@ def export(self, request, pk):
         # Not using existing get_object method because it requires access to the organization user attribute
         export = ical_export_from_schedule(self.request.auth.schedule)
         return Response(export, status=status.HTTP_200_OK)
+
+    @action(methods=["get"], detail=True)
+    def oncall_shifts_export(self, request, pk):
+        schedule = self.get_object()
+
+        if not isinstance(schedule, OnCallScheduleWeb):
+            return Response(
+                "OnCall shifts exports are currently only available for web calendars",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        start_date_field_name = "start_date"
+        end_date_field_name = "end_date"
+
+        def _field_is_required(field_name):
+            return Response(f"{field_name} is required", status=status.HTTP_400_BAD_REQUEST)
+
+        def _field_is_invalid_date(field_name):
+            return Response(
+                f"{field_name} is not a valid date, must be in any valid ISO 8601 format",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        def _convert_date(value):
+            if not value:
+                return None
+
+            try:
+                return date.fromisoformat(value)
+            except ValueError:
+                return None
+
+        start_date_str = request.query_params.get(start_date_field_name, None)
+        start_date = _convert_date(start_date_str)
+
+        end_date_str = request.query_params.get(end_date_field_name, None)
+        end_date = _convert_date(end_date_str)
+
+        if start_date_str is None:
+            return _field_is_required(start_date_field_name)
+        elif start_date is None:
+            return _field_is_invalid_date(start_date_field_name)
+        elif end_date_str is None:
+            return _field_is_required(end_date_field_name)
+        elif end_date is None:
+            return _field_is_invalid_date(end_date_field_name)
+
+        if start_date > end_date:
+            return Response(
+                f"{start_date_field_name} must be less than or equal to {end_date_field_name}",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
@grafana/grafana-oncall-backend any suggestions on a cleaner way to do this validation? it works well, but I feel like there is a simpler way w/ `django-rest-framework`